omnigibson/utils/object_state_utils.py

```diff
@@ -298,7 +300,7 @@ def calculate_projection_area_and_diagonal(obj, dims):
         diagonal (float): diagonal of the convex hull of the projected points
     """
     cloth = obj.links["base_link"]
-    points = cloth.particle_positions[:, dims]
+    points = cloth.get_particle_positions(keypoints_only=True)[:, dims]
```

Comment:
We can get rid of the following code, right?

```

    if points.shape[0] > m.N_POINTS_CONVEX_HULL:
        # If there are too many points, subsample m.N_POINTS_CONVEX_HULL deterministically for efficiency purpose
        np.random.seed(0)
        random_idx = np.random.randint(0, points.shape[0], m.N_POINTS_CONVEX_HULL)
        points = points[random_idx]
```

Reply:
yep, done!